botcity/maestro/sdk.py

```diff
@@ -1067,6 +1070,50 @@ def create_credential(self, label: str, key: str, value):
             if not req.ok:
                 req.raise_for_status()
 
+    @since_version("3.0.0")
+    @ensure_access_token()
+    def update_credential(self, label: str, key: str, new_value):
+        """Update a credential with a new value
+        Args:
+            label: Credential set name
+            key: Key name within the credential set
+            new_value: Credential new value
+        """
+        credential = self._get_credential_by_label(label=label)
+        if credential is None:
+            raise ValueError(f"Credential set '{label}' does not exists.")
+        secrets = credential.get("secrets", [])
+
+        index = next((i for i, d in enumerate(secrets) if d.get('key') == key), None)
+        if index is None:
+            raise ValueError(f"Key '{key}' does not exist in credential set '{label}'.")
```

Comment:
What do you think about creating another function to get the index from the secrets of the credential? This way, we also isolate this case in a single place. For example:

```python
	def _get_index_by_secrets(self, secrets: list):
		index = next((i for i, d in enumerate(secrets) if d.get('key') == key), None)
		if index is None:
		    raise ValueError(f"Key '{key}' does not exist in credential set '{label}'.")
		
		return index
```

Reply:
Addressed via https://github.com/botcity-dev/botcity-maestro-sdk-python/pull/41/commits/522ebc5d4fa3b58b0acc309ffa6237c4ff03a001